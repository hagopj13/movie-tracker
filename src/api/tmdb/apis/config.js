// @flow
import axios from '../axiosInstance';

export const getConfig = () => axios.get('/configuration');

export const getAllGenres = () => axios.get('/genre/movie/list');
