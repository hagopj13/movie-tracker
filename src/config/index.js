import React from 'react';
import DiscoverIcon from '@material-ui/icons/Movie';
import UpcomingIcon from '@material-ui/icons/Update';
import ProfileIcon from '@material-ui/icons/Person';

export const navigationItems = [
  {
    id: 'DISCOVER',
    label: 'Discover',
    to: '/',
    icon: <DiscoverIcon />,
    requiresAuth: false,
  },
  {
    id: 'UPCOMING',
    label: 'Upcoming',
    to: '/upcoming',
    icon: <UpcomingIcon />,
    requiresAuth: false,
  },
  {
    id: 'PROFILE',
    label: 'Profile',
    to: '/profile',
    icon: <ProfileIcon />,
    requiresAuth: true,
  },
];

export const dialogTypes = {
  LOGIN: 'LOGIN',
};

export const sortOptions = [
  { value: 'popularity.desc', label: 'Popularity Descending' },
  { value: 'popularity.asc', label: 'Popularity Ascending' },
  { value: 'vote_average.desc', label: 'Rating Descending' },
  { value: 'vote_average.asc', label: 'Rating Ascending' },
  { value: 'release_date.desc', label: 'Release Date Descending' },
  { value: 'release_date.asc', label: 'Release Date Ascending' },
];
