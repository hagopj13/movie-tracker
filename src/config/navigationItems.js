import React from 'react';
import DiscoverIcon from '@material-ui/icons/Movie';
import UpcomingIcon from '@material-ui/icons/Update';
import ProfileIcon from '@material-ui/icons/Person';

export default [
  {
    id: 'DISCOVER',
    label: 'Discover',
    to: '/',
    icon: <DiscoverIcon />,
    requiresAuth: false,
  },
  {
    id: 'UPCOMING',
    label: 'Upcoming',
    to: '/upcoming',
    icon: <UpcomingIcon />,
    requiresAuth: false,
  },
  {
    id: 'PROFILE',
    label: 'Profile',
    to: '/profile',
    icon: <ProfileIcon />,
    requiresAuth: true,
  },
];
