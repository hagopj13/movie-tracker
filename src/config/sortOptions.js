export default [
  { value: 'popularity.desc', label: 'Popularity Descending' },
  { value: 'popularity.asc', label: 'Popularity Ascending' },
  { value: 'vote_average.desc', label: 'Rating Descending' },
  { value: 'vote_average.asc', label: 'Rating Ascending' },
  { value: 'release_date.desc', label: 'Release Date Descending' },
  { value: 'release_date.asc', label: 'Release Date Ascending' },
];
