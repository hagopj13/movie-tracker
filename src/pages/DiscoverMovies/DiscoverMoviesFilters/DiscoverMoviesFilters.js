// @flow
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import type { Moment } from 'moment';

import MoviesFilters from 'components/Movies/MoviesFilters/MoviesFilters';
import discoverActions from 'store/discover/discover.actions';
import discoverSelectors from 'store/discover/discover.selectors';
import configSelectors from 'store/config/config.selectors';
import type { State as FiltersState } from 'store/common/filters/filters.reducer';
import type { Genre } from 'store/config/config.reducer';

type Props = {
  allGenres: Genre[],
  selectedFilters: FiltersState,
  onUpdateResults: () => void,
  onSetSortBy: (sortBy: string) => void,
  onToggleGenre: (genreId: number) => void,
  onSetReleaseDateStart: (releaseDateStart: Moment) => void,
  onSetReleaseDateEnd: (releaseDateEnd: Moment) => void,
};

const DiscoverMoviesFilters = (props: Props) => {
  const {
    allGenres,
    selectedFilters,
    onUpdateResults,
    onSetSortBy,
    onToggleGenre,
    onSetReleaseDateStart,
    onSetReleaseDateEnd,
  } = props;

  return (
    <MoviesFilters
      allGenres={allGenres}
      selectedFilters={selectedFilters}
      onFiltersChanged={onUpdateResults}
      onSetSortBy={onSetSortBy}
      onToggleGenre={onToggleGenre}
      onSetReleaseDateStart={onSetReleaseDateStart}
      onSetReleaseDateEnd={onSetReleaseDateEnd}
    />
  );
};

const mapStateToProps = createStructuredSelector({
  allGenres: configSelectors.selectAllGenres,
  selectedFilters: discoverSelectors.selectFilters,
});

const mapDispatchToProps = {
  onUpdateResults: discoverActions.fetchMovies,
  onSetSortBy: discoverActions.setSortBy,
  onToggleGenre: discoverActions.toggleGenre,
  onSetReleaseDateStart: discoverActions.setReleaseDateStart,
  onSetReleaseDateEnd: discoverActions.setReleaseDateEnd,
};

export default connect(mapStateToProps, mapDispatchToProps)(DiscoverMoviesFilters);
