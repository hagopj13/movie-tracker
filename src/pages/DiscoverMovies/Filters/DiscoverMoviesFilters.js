// @flow
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import type { Moment } from 'moment';

import MovieFilters from 'components/Movie/Filters/MovieFilters';
import discoverActions from 'store/discover/discover.actions';
import discoverSelectors from 'store/discover/discover.selectors';
import configSelectors from 'store/config/config.selectors';
import type { State as FiltersState } from 'store/common/filters/filters.reducer';
import type { Genre } from 'types';

type Props = {
  allGenres: Genre[],
  selectedFilters: FiltersState,
  onUpdateResults: () => void,
  onSetSortBy: (sortBy: string) => void,
  onToggleGenre: (genreId: number) => void,
  onSetReleaseDateStart: (releaseDateStart: Moment) => void,
  onSetReleaseDateEnd: (releaseDateEnd: Moment) => void,
};

export const DiscoverMoviesFilters = (props: Props) => {
  const {
    allGenres,
    selectedFilters,
    onUpdateResults,
    onSetSortBy,
    onToggleGenre,
    onSetReleaseDateStart,
    onSetReleaseDateEnd,
  } = props;

  return (
    <MovieFilters
      allGenres={allGenres}
      selectedFilters={selectedFilters}
      onFiltersChanged={onUpdateResults}
      onSetSortBy={onSetSortBy}
      onToggleGenre={onToggleGenre}
      onSetReleaseDateStart={onSetReleaseDateStart}
      onSetReleaseDateEnd={onSetReleaseDateEnd}
    />
  );
};

const mapStateToProps = createStructuredSelector({
  allGenres: configSelectors.selectAllGenres,
  selectedFilters: discoverSelectors.selectFilters,
});

const mapDispatchToProps = {
  onUpdateResults: discoverActions.fetchMovies,
  onSetSortBy: discoverActions.setSortBy,
  onToggleGenre: discoverActions.toggleGenre,
  onSetReleaseDateStart: discoverActions.setReleaseDateStart,
  onSetReleaseDateEnd: discoverActions.setReleaseDateEnd,
};

export default connect(mapStateToProps, mapDispatchToProps)(DiscoverMoviesFilters);
