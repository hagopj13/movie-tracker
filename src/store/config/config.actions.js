// @flow
import type { ImagesConfig, Genre } from 'types';

import ConfigActionTypes from './config.types';

const fetchImagesConfig = () => ({
  type: ConfigActionTypes.FETCH_IMAGES_CONFIG,
});

const fetchImagesConfigStart = () => ({
  type: ConfigActionTypes.FETCH_IMAGES_CONFIG_START,
});

const fetchImagesConfigSuccess = (imagesConfig: ImagesConfig) => ({
  type: ConfigActionTypes.FETCH_IMAGES_CONFIG_SUCCESS,
  payload: { imagesConfig },
});

const fetchImagesConfigFailure = (error: string) => ({
  type: ConfigActionTypes.FETCH_IMAGES_CONFIG_FAILURE,
  payload: { error },
});

const fetchAllGenres = () => ({
  type: ConfigActionTypes.FETCH_ALL_GENRES,
});

const fetchAllGenresStart = () => ({
  type: ConfigActionTypes.FETCH_ALL_GENRES_START,
});

const fetchAllGenresSuccess = (genres: Genre[]) => ({
  type: ConfigActionTypes.FETCH_ALL_GENRES_SUCCESS,
  payload: { genres },
});

const fetchAllGenresFailure = (error: string) => ({
  type: ConfigActionTypes.FETCH_ALL_GENRES_FAILURE,
  payload: { error },
});

export default {
  fetchImagesConfig,
  fetchImagesConfigStart,
  fetchImagesConfigSuccess,
  fetchImagesConfigFailure,
  fetchAllGenres,
  fetchAllGenresStart,
  fetchAllGenresSuccess,
  fetchAllGenresFailure,
};
