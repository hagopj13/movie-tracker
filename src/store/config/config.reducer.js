// @flow
import { handleActions } from 'redux-actions';

import type { ImagesConfig, Genre } from 'types';

import ConfigActionTypes from './config.types';

type State = {
  images: ImagesConfig,
  genres: Genre[],
};

export const initialState: State = {
  images: {
    baseImageUrl: '',
    backdropSize: '',
    posterSize: '',
    profileSize: '',
  },
  genres: [],
};

const fetchImagesConfigSuccess = (state: State, action: any) => ({
  ...state,
  images: action.payload.imagesConfig,
});

const fetchAllGenresSuccess = (state: State, action: any) => ({
  ...state,
  genres: action.payload.genres,
});

const actionHandler = {
  [ConfigActionTypes.FETCH_IMAGES_CONFIG_SUCCESS]: fetchImagesConfigSuccess,
  [ConfigActionTypes.FETCH_ALL_GENRES_SUCCESS]: fetchAllGenresSuccess,
};

export default handleActions(actionHandler, initialState);
