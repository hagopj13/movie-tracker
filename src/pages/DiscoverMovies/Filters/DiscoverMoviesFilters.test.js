import React from 'react';
import moment from 'moment';
import { createShallow } from '@material-ui/core/test-utils';

import { genres } from 'store/fixtures/config';
import sortOptions from 'config/sortOptions';

import { DiscoverMoviesFilters } from './DiscoverMoviesFilters';

describe('DiscoverMoviesFilters component', () => {
  let shallow;
  let allGenres;
  let selectedFilters;
  let wrapper;

  const mockUpdateResults = jest.fn();
  const mockSetSortBy = jest.fn();
  const mockToggleGenre = jest.fn();
  const mockSetReleaseDateStart = jest.fn();
  const mockSetReleaseDateEnd = jest.fn();

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    jest.clearAllMocks();
    allGenres = genres;
    selectedFilters = {
      sortBy: sortOptions[1].value,
      genres: [genres[1].id],
      releaseDateStart: moment('2020-01-01'),
      releaseDateEnd: moment('2020-02-01'),
    };
    const mockProps = {
      allGenres,
      selectedFilters,
      onUpdateResults: mockUpdateResults,
      onSetSortBy: mockSetSortBy,
      onToggleGenre: mockToggleGenre,
      onSetReleaseDateStart: mockSetReleaseDateStart,
      onSetReleaseDateEnd: mockSetReleaseDateEnd,
    };
    wrapper = shallow(<DiscoverMoviesFilters {...mockProps} />);
  });

  it('should render the DiscoverMoviesFilters component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
