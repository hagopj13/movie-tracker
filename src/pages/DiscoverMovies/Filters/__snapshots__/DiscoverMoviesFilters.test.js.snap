// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscoverMoviesFilters component should render the DiscoverMoviesFilters component correctly 1`] = `
<MovieFilters
  allGenres={
    Array [
      Object {
        "id": 28,
        "name": "Action",
      },
      Object {
        "id": 12,
        "name": "Adventure",
      },
      Object {
        "id": 16,
        "name": "Animation",
      },
      Object {
        "id": 35,
        "name": "Comedy",
      },
    ]
  }
  onFiltersChanged={[MockFunction]}
  onSetReleaseDateEnd={[MockFunction]}
  onSetReleaseDateStart={[MockFunction]}
  onSetSortBy={[MockFunction]}
  onToggleGenre={[MockFunction]}
  selectedFilters={
    Object {
      "genres": Array [
        12,
      ],
      "releaseDateEnd": "2020-01-31T23:00:00.000Z",
      "releaseDateStart": "2019-12-31T23:00:00.000Z",
      "sortBy": "popularity.asc",
    }
  }
/>
`;
